src/MultiDimFit.cc

```diff
@@ -183,7 +183,13 @@ bool MultiDimFit::runSpecific(RooWorkspace *w, RooStats::ModelConfig *mc_s, RooS
     if(w->var("r")) {w->var("r")->Print();}
     if ( loadedSnapshot_ || res.get() || keepFailures_) {
         for (int i = 0, n = poi_.size(); i < n; ++i) {
-            poiVals_[i] = poiVars_[i]->getVal();
+            RooAbsArg *rfloat = (*res).floatParsFinal().find(poi_[i].c_str());
```

Comment:
Think it's possible that `res` can be null here, e.g. if we've loaded a snapshot the initial fit may not have been performed. It should also be safe to always set the poiVals_ from the RooFitResult preferentially if it exists, not only when algo is Singles